Makefile.work

```diff
@@ -115,6 +115,14 @@ OVERLAY_MODULE_CHECK := \
 
 BUILD_TIMESTAMP := $(shell date +%Y%m%d\.%H%M%S)
 
+# Create separate Docker lockfiles for saving vs. loading an image.
+ifeq ($(DOCKER_LOCKDIR),)
+override DOCKER_LOCKDIR := /tmp/docklock
+endif
+DOCKER_LOCKFILE_SAVE := $(DOCKER_LOCKDIR)/docker_save.lock
+$(shell mkdir -m 0777 -p $(DOCKER_LOCKDIR))
+$(shell [ -f $(DOCKER_LOCKFILE_SAVE) ] || (touch $(DOCKER_LOCKFILE_SAVE) && chmod 0777 $(DOCKER_LOCKFILE_SAVE)))
```

Comment:
i understand the user name part why it is needed.

but i do not understand the docker_locker dir, can you explain why this mechanism is introduced?

Reply:
Lock is required when multiple builds are triggered on the same build server.  Save and Load of docker image should be protected with lock as two different builds can try to load/save the docker image with the same name.

Reply:
I thought you changed to different names

Reply:
Do you really need a name change?

Reply:
my question was since you append user name to the docker image name, why is it still an issue when multiple users trying to build? why is the lock still needed?

Reply:
During the docker image save/load, it should be saved/loaded without username extension as this is directly used in the sonic installer image. 
Eg. 
docker-database:latest 

But during the build, it should be tagged as username extension as multiple builds can happen in parallel. 
Eg:
docker-database-<user>:latest

Reply:
when you say load/save, are you referring to the process of building docker phase? in the build process, why not keep the user name to do load and save, and only at the build_debian.sh phase to load the image into the image without username?

Reply:
The docker image on the  docker-xxx.gz image file should have an image tag as latest because individual docker images can be upgraded on the device and we cant use the user-specific extension on docker image which causes the docker image upgrade failure on the device.